src/mem/freelist.h

```diff
@@ -71,11 +71,14 @@ namespace snmalloc
    */
   class FreeObject
   {
+    template<bool, bool>
+    friend class FreeListBuilder;
+
     union
     {
-      CapPtr<FreeObject, CBAlloc> next_object;
+      FreeObject* next_object;
```

Comment:
Why can these be bare pointers?  Don't they have to be wild (except on systems with CHERI + MTE)?